File: python/euclid.py
def gcd(a, b):
    while (b > 0):
        a, b = b, a % b
    
    return a

def extgcd(a, b):
    # Return in form a*x + b*y
    # x is the inverse of a
    # y is the inverse if b
    # There is only an inverse if gcd(a,b) == 1
    if a == 0:
        return (0, 1)
    else:
        y, x = extgcd(b % a, a)
        return ((x - (b // a) * y) % b, y)

def moduloInverseOf(a, b):
    inv = extgcd(a, b)
    return inv[0]

if __name__ == "__main__":
    
    a = int(input("Enter a: "))
    b = int(input("Enter b: "))

    resultGCD = gcd(a,b)
    
    inverse = "No Inverse"
    if resultGCD == 1:
        inverse = moduloInverseOf(a, b)

    print("Greatest Common Divisor: ", resultGCD)
    print("Inverse of " + str(a) + " mod " + str(b) + " is: ", inverse)

File: C++/euclid.cpp
#include "euclid.hpp"

int Euclid::gcd(int a, int b) {
    
    int tmp;
    while (b > 0) {
        tmp = b;
        b = a % b;
        a = tmp;
    }

    return a;
}

std::tuple<int, int> Euclid::extgcd(int a, int b) {
    // Return in x and y form a*x + b*y
    // x is the inverse of a
    // y is the inverse if b
    // There is only an inverse if gcd(a,b) == 1
    if (a == 0) {
        return std::make_tuple(0, 1);
    }else {
        std::tuple<int, int> result = extgcd(b % a, a);

        int x1 = std::get<1>(result) - int(b / a) * std::get<0>(result);
        int y1 = std::get<0>(result);

        return std::make_tuple(x1, y1);
    }
}

int Euclid::mod(int a, int b) {
    return ((a%b)+b) % b;
}

int Euclid::moduloInverseOf(int a, int b) {
    std::tuple<int, int> inv = extgcd(a, b);
    return mod(std::get<0>(inv), b);
}